lib/ansible/module_utils/basic.py

```diff
@@ -1332,6 +1340,10 @@ def set_directory_attributes_if_different(self, file_args, changed, diff=None, e
     def set_file_attributes_if_different(self, file_args, changed, diff=None, expand=True):
         return self.set_fs_attributes_if_different(file_args, changed, diff, expand)
 
+    def add_atomic_move_warnings(self):
+        for path in self._created_files:
+            self.warn("File '%s' created with default permissions. Specify 'mode' to avoid this warning." % to_native(path))
```

Comment:
I'd include the default permissions in the message, as well as a brief note that they are not consistent across versions. Since that's likely a rather verbose message, perhaps it would be better to link to a docs page explaining the situation instead?

Reply:
I was planning to do that just needed to figure out the string format foo. Turns out it's pretty easy: `{:o}` gives me what I want.